### ISSUE TITLE
debuging kernel doesn't start: invalid `Elf' handle

### ISSUE THREAD

**Issue by Nudin** [osandov/drgn#10]
I followed the install-guide. After that I started it via `sudo drgn -k` and got:
```
Traceback (most recent call last):
  File "/usr/bin/drgn", line 11, in <module>
    load_entry_point('drgn==0.0.1', 'console_scripts', 'drgn')()
  File "/usr/lib/python3.7/site-packages/drgn-0.0.1-py3.7-linux-x86_64.egg/drgn/internal/cli.py", line 90, in main
    prog.load_default_debug_info()
_drgn.FileFormatError: libelf error: invalid `Elf' handle
```

Any idea what I'm doing wrong?

**Comment by osandov** [comment]
Could you provide the output of running `sudo strace drgn -k`? Hopefully I can figure out which file failed from that.

Some other general information that would be appreciated:

* What distro and distro version is this on?
* What version of elftutils do you have installed? `eu-readelf --version` should give you that unless you did anything special to get the elfutils libraries.

**Comment by Nudin** [comment]
- Arch Linux
- eu-readelf (elfutils) 0.176
- strace: https://gist.github.com/Nudin/37459465d89db7ef21dc1f217519d752

**Comment by osandov** [comment]
Ah, it looks like you're trying to debug the official Arch kernel, but Arch doesn't ship debugging information. Unfortunately, the only way to get debugging information on Arch is to build the kernel yourself with `CONFIG_DEBUG_INFO=y`. However, drgn should definitely be giving you a better error message here. I can reproduce the issue here, so I'll fix that.

**Comment by Nudin** [comment]
Makes total sense! Adding a proper error message sounds the right way! Thanks for the quick help.

**Comment by osandov** [comment]
This is fixed by the larger rework in 423d2cd5009e562b96e1fbc0d32a49d056222485:
```
$ sudo drgn -k
could not get debugging information for:
/usr/lib/modules/5.3.1-arch1-1-ARCH/kernel/drivers/input/serio/serio.ko.xz (libdwfl error: No DWARF information found)
/usr/lib/modules/5.3.1-arch1-1-ARCH/kernel/drivers/input/serio/i8042.ko.xz (libdwfl error: No DWARF information found)
/usr/lib/modules/5.3.1-arch1-1-ARCH/kernel/drivers/net/net_failover.ko.xz (libdwfl error: No DWARF information found)
/usr/lib/modules/5.3.1-arch1-1-ARCH/kernel/drivers/net/virtio_net.ko.xz (libdwfl error: No DWARF information found)
/usr/lib/modules/5.3.1-arch1-1-ARCH/kernel/drivers/macintosh/mac_hid.ko.xz (libdwfl error: No DWARF information found)
... 40 more
```
Thanks for reporting this!
